integration/spark2/src/main/spark2.2/CarbonSessionState.scala

```diff
@@ -121,6 +123,19 @@ class CarbonSessionCatalog(
         toRefreshRelation = refreshRelationFromCache(name, carbonDatasourceHadoopRelation)
       case LogicalRelation(carbonDatasourceHadoopRelation: CarbonDatasourceHadoopRelation, _, _) =>
         toRefreshRelation = refreshRelationFromCache(name, carbonDatasourceHadoopRelation)
+      case SubqueryAlias(_, relation) if
+      relation.getClass.getName.equals("org.apache.spark.sql.catalyst.catalog.CatalogRelation") ||
+      relation.getClass.getName.equals("org.apache.spark.sql.catalyst.catalog.HiveTableRelation") ||
+      relation.getClass.getName.equals(
+        "org.apache.spark.sql.catalyst.catalog.UnresolvedCatalogRelation") =>
+        val catalogTable = CarbonReflectionUtils.getFieldOfCatalogTable("tableMeta",
+          relation).asInstanceOf[CatalogTable]
+        carbonEnv.carbonMetastore.checkSchemasModifiedTimeAndReloadTable(catalogTable.identifier)
```

Comment:
In case the table has different timestamp and removed from metadata through checkSchemasModifiedTimeAndReloadTable, then only call refresh table.